### PR TITLE
Adds the mobile operating table.

### PR DESCRIPTION
**Pull request by DrSnips**
Surgery tables while useful can't really be moved around and require you to have a decent set of tools on you in order to pick up and move it. While normal roller beds _can_ act like surgery tables it's only a chance to work. This introduces a way to make roller beds as well as making them a bit more reliable for surgery.
These could be extremely helpful for doctors who need to keep working on someone while they need to move around. IE the escape shuttle or medical when some parts of it become hazardous. While it does act like a mobile table you still need to stand still while working on someone.
This is considered a roller bed so medical borgs can hold them in their storage.
fm
M.O.T. sprite.
:cl:
 * rscadd: Added the mobile operating table. It works just like a roller bed but allows you to to surgery on it without the chance of failure like a normal roller bed. Can be made in the protolathe or exosuit fab under the misc tabs.

### CODE REVIEW

**Review by PJB3005 — APPROVED** [review]

**Comment by Pathid** [comment]
Just a suggestion here: Have an idle sprite for the cardiogram for when the patient has no pulse. The regular operating table does this.

**Comment by DrSnips** [comment]
I will see what I can do since a normal table is slightly different than a bed.

**Comment by DrSnips** [comment]
I will throw in the up idle sprite if someone wants to take that on later but I'm satisfied with the monitor being purely aesthetic for now.

**Comment by Exxion** [comment]
What are the research requirements?
And why can it be made in the mechfab?

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 1]
```diff
@@ -253,6 +253,22 @@
...
 
+//An advanced roller bed that allows you to do surgery on it 100% of the time in place of the 75% chance of the normal one.
+/obj/item/roller/adv
+	name = "mobile operating table"
+	desc = "A collapsed mobile operating table that can be carried around."
+	icon = 'icons/obj/rollerbed.dmi'
+	icon_state = "adv_folded"
+	bed_type = /obj/structure/bed/roller/adv
+
+/obj/structure/bed/roller/adv
```
Comment (ultraplen) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: a more descriptive path than adv would be nice

**Comment by DrSnips** [comment]
Needs bio 5, engineering 4, and programming/data 2. The cost is similar to surgeon switchtools and janicart upgrades at 10000 metal and 5000 glass. I saw it as something similar to janicart upgrades so I put it in exosuit fabs as well.

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm (hunk 1 above)
Comment (DrSnips) [reply code/game/objects/structures/stool_bed_chair_nest/bed.dm]: Sure thing. Somthing like /roller/surgery ok?

**Pull request merged**